src/shared/ui/modal-window/modalWindow.vue

```diff
@@ -0,0 +1,38 @@
+<script setup lang="ts">
+import { modal, modalClose, open, focusTrap } from '@/shared/lib/modal'
+import './modal-window.scss'
+import '@/shared/styel/helpers.scss'
+</script>
+
+<template>
+  <div
+    class="dialog-bg"
+    :class="{ open: open }"
+    :aria-hidden="!open"
+    @click="modalClose()"
+    @keydown.esc="modalClose()"
+  >
+    <dialog class="modal" :open="open" @click.stop>
+      <header class="modal__header">
+        <h3 id="modal-title" class="modal__title">
+          {{ modal.name }}
+        </h3>
+        <button
+          class="modal__button"
+          aria-label="закрыть модальное окно"
+          @click="modalClose()"
+        >
+          X
+        </button>
+      </header>
+      <article class="modal__article">
+        <component :is="modal.component" />
+      </article>
+      <input
+        class="modal__end-tab visually-hidden"
+        type="text"
+        @focus="focusTrap()"
```

Comment:
Не вызываем функцию.

```suggestion
        @focus="focusTrap"
```